cranelift/codegen/src/opts/selects.isle

```diff
@@ -4,6 +4,27 @@
 (rule (simplify (select    ty _ x x)) x)
 (rule (simplify (bitselect ty _ x x)) x)
 
+;; if icmp(x, y) { 1 } else { 0 } => icmp(x, y)
+(rule (simplify (select $I8 cmp@(icmp _ cc x y)
+                        (iconst _ (u64_from_imm64 1))
+                        (iconst _ (u64_from_imm64 0))))
+    cmp)
+;; if icmp(x, y) { 0 } else { 1 } => !icmp(x, y)
+(rule (simplify (select $I8 (icmp ty cc x y)
+                        (iconst _ (u64_from_imm64 0))
+                        (iconst _ (u64_from_imm64 1))))
+    (icmp ty (intcc_complement cc) x y))
+;; if icmp(x, y) { 1 } else { 0 } => uextend(icmp(x, y))
+(rule (simplify (select ty cmp@(icmp _ cc x y)
+                        (iconst _ (u64_from_imm64 1))
+                        (iconst _ (u64_from_imm64 0))))
+    (uextend ty cmp))
+;; if icmp(x, y) { 0 } else { 1 } => uextend(!icmp(x, y))
+(rule (simplify (select sty (icmp cty cc x y)
+                        (iconst _ (u64_from_imm64 0))
+                        (iconst _ (u64_from_imm64 1))))
+    (uextend sty (icmp cty (intcc_complement cc) x y)))
```

Comment:
To further expand on this a bit as well, what you've defined here  are effectively two overlapping rules, for example the rule on line 23 (this one) and the rule on line 13 (the i8 case above). For an i8-producing-select both rules match, and I believe ISLE allows that for the "multi" constructor that `simplify` is (since I believe it's generating all matches and yielding them to the egraph driver). While the above rule is "lower cost" and will get selected eventually I think it's probably still best to avoid the possibility of accidentally creating invalid CLIF